doc/developer-guides/coding_guidelines.rst

```diff
@@ -0,0 +1,1000 @@
+.. _coding_guidelines:
+
+Coding Guidelines
```

Comment:
Some ideas are borrowed from the MISRA-C standard, but all descriptions in this doc is written by a developer without access to the standard to avoid unintended plagiarism.